text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,111 @@
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0000](https://github.com/casperlabs/ceps/pull/0000)
+
+This CEP describes a proposal for the process of handling equivocation catastrophes in Highway consensus (or, more generally: handling situations that break the assumptions behind the security of the consensus algorithm being used).
+
+## Motivation
+
+[motivation]: #motivation
+
+Every consensus algorithm's safety guarantees rely on some assumptions being upheld. In the case of Highway, the main such assumption is about the validators not equivocating, that is: not producing conflicting votes. The algorithm can withstand some amount of equivocations, but if the total weight of equivocating validators exceeds a threshold called the Fault Tolerance Threshold (FTT), they could revert finalized blocks, which could lead for example to double spending. We call such a situation an "equivocation catastrophe" and it is something that cannot be handled automatically (as it would effectively contradict well-established results about the limitations of consensus algorithms in general). Thus, we require some kind of an external mechanism for dealing with such situations, should they ever happen.
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The procedure for handling the equivocation catastrophes will have to consist of two main parts: what the software is expected to do in such a case, and what are the manual actions required to proceed.
+
+The obvious first thing that should happen in the case of an equivocation catastrophe is that the software should alert the users about the situation. The node software is aware of the FTTs of the blocks, so when it detects equivocations, it can notice if the total weight of the equivocators exceeds the FTT of a block. When the FTT of any block is exceeded, the users should be made aware of it.
+
+In order to make sure that as many validators as possible become aware of all the equivocations eventually, the operation of the nodes shouldn't stop. Finalizing blocks should be halted, as finalization ceases to mean anything when it can be reverted, but the nodes should continue gossiping votes.
+
+The nodes could also output the hash of the last block whose FTT hasn't been exceeded and all the transactions included in the later blocks, and the list of equivocators. These will be useful for the manual steps.
+
+The manual actions afterwards will consist of two main steps:
+
+1. Reaching social consensus about the block to be used as the new starting point,
+2. Restarting the network with the new starting point.
+
+We will go into more detail about these steps below.
+
+### Social consensus
+
+For the initial version of this CEP, we assume that we have some kind of platform that enables open communication between validators and CasperLabs. The exact details of what that platform will be (e-mail? Discord? a forum?) can be left to be determined in the future.
```

Comment:
We should not even mention Discord. Possible options are email with signatures, or a forum/Matrix instance not controlled by a single validator.

The chosen platform should allow for storing, linking, referencing and displaying the whole public process and communication. We also should make sure that the content is in a format that can be displayed elsewhere. We should consider that people would refer back to these processes and disputes for the next 100 years.